### DISCUSSION TITLE
restore save file

### DISCUSSION THREAD

**Discussion by rgerhyejhsrgerhjhr** [rafaelvcaetano/melonDS-android#1172] [Support]
hello guys, my melonds didn't save progress. I think I found the save file (check image) but don't know what to do now. Could any1 navigate me please?

![image](https://github.com/rafaelvcaetano/melonDS-android/assets/135761983/1d2ab683-6f16-4367-b367-2004e63d0b6c)

**Comment by rgerhyejhsrgerhjhr** [comment]
this appeared on my apps front page

**Comment by rafaelvcaetano** [comment]
To use that save file, you just need to make sure that the ROM file name matches the save file name and that they are in the same directory. For your specific case, the ROM file name should be "Pokemon Platinum Version (US)(XenoPhobia).nds".

If this still doesn't work, go to Settings > Save Files and make sure that the first option (called "Save next to ROM file") is enabled.

**Reply by rgerhyejhsrgerhjhr** [reply]
Hey! I do have another safe file for the previous 30 hours I played so those 2 save files exist at the same time. Just now it started not working anymore. Is there anything I need to consider because of that? Oh and thanks for your reply!

**Reply by rgerhyejhsrgerhjhr** [reply]
what I mean by that is that it only always loads the old save file and not the new one. That is especially odd because the difference between old save and new save were like 6h of playing with saving and logging off several time during those 6h

**Reply by rgerhyejhsrgerhjhr** [reply]
Do you also know how I can find the directory? I basically need to copy the save and paste it into the directory in which "Pokemon Platinum Version (US)(XenoPhobia).nds" also is, correct? I am sorry if I am asking dumb questions but I am really not good at computer stuff

**Reply by rafaelvcaetano** [reply]
The emulator will only load the save file whose name matches the ROM file name and has the ".sav" extension. Since 2 files cannot have the same name, you need to confirm which one has the correct name and change the names if you switch them. So, you can have as many save files as you want, but the emulator will only use the one that has the correct name.

**Reply by rafaelvcaetano** [reply]
If the option "Save next to ROM file" is enabled, then the files are in the same directory that has the ROMs. To see the directory is being used to search for ROMs, go to Settings > ROMs and the directory path is displayed under the "Search directory " option.

**Reply by rgerhyejhsrgerhjhr** [reply]
OH. I just checked and there are two save files as I said but the one that always loads has a "3541 - " before Pokemon Platinum...
and that also applies to the .nds   .So my only job is to rename the files or do I also need to move stuff around?

**Reply by rgerhyejhsrgerhjhr** [reply]
Under the search directory I only found the correct (newer) save file and the ZIP data, but not the .nds and neither the old save file. I do have the "Save next to ROM file" enabled

**Comment by rgerhyejhsrgerhjhr** [comment]
okay. Now my zip and nds files are in the same directory as my correct save file and all that did for me is that I now have one .zip and one .nds option when I run melonds and both run the wrong save file

**Reply by rafaelvcaetano** [reply]
You don't need both the zip and nds files. You just need one.

Since you are still having issues, could you send me screenshots of the ROM and save files?

**Reply by rgerhyejhsrgerhjhr** [reply]
yes of course, gimme a second

**Reply by rgerhyejhsrgerhjhr** [reply]
![image](https://github.com/rafaelvcaetano/melonDS-android/assets/135761983/0e00c512-4462-4664-997b-f89b24272b5b)
![image](https://github.com/rafaelvcaetano/melonDS-android/assets/135761983/9fc18939-0fa1-4a89-a1bf-f96646c1a535)
![image](https://github.com/rafaelvcaetano/melonDS-android/assets/135761983/f17d26f7-39cf-44f9-89e7-0a9cd858678d)

**Reply by rafaelvcaetano** [reply]
And which save file do you want to use? The emulator should be loading the save file ending in ".sav". The ".dsv" file has a different name and that file extension is not supported. If you want to use that .dsv file, you need to convert it to .sav. Here is a tool that allows that: https://www.save-editor.com/tools/wse_ds_save_converter_for_emulator_desmume_dsv.html

**Reply by rgerhyejhsrgerhjhr** [reply]
> And which save file do you want to use? The emulator should be loading the save file ending in ".sav". The ".dsv" file has a different name and that file extension is not supported. If you want to use that .dsv file, you need to convert it to .sav. Here is a tool that allows that: https://www.save-editor.com/tools/wse_ds_save_converter_for_emulator_desmume_dsv.html

I think the dsv file is the wrong one (older one) since the date is 28.05, which I suppose is the last time I played pokemon platinum before today. But feel free to correct me here. I wish to play the one that has the progress I made today as well

**Reply by rafaelvcaetano** [reply]
Sorry, I didn't notice the .dsv file was the older one. In that case, I am a bit confused. Is that .sav file the one that has all the progress? Or is there another .sav file that is not in this folder? 

If you have only this one, and it is missing those 6 hours of progress that you said, then I am afraid there's nothing that can be done to restore that progress :( And if you say that you had multiple sessions between those 6 hours of progress, then it can't be the emulator's fault. Maybe you loaded an old save state and the saved the game, overwriting the newer save?

**Reply by rgerhyejhsrgerhjhr** [reply]
Idk to be honest, I only ever saved within the game and nothing else. Is there any chance that my device messed up the times? Meaning the .dsv is newer? Is there any chance for that? Maybe does the game still load the .dsv in case it is older? Can I somehow look into the files and see their progress?

**Reply by rafaelvcaetano** [reply]
I don't believe the device could have done that. But yes, you can test the dsv file. First, change the name of the .sav file so that it doesn't match the name of the ROM. Then, use the tool I indicated earlier to convert it to .sav. Finally, make sure the name of the new .sav file matches the name of the ROM.

To revert, change the names of the files so that the original .sav file has the same name as the ROM.

**Reply by rgerhyejhsrgerhjhr** [reply]
I tested the file and unfortunately it was even older and from another emulator and didn't manage to get transferred. Guess my progress is gone for good. I still appreciate your help, thank you very much!
